### PR TITLE
feature/add-loader-and-rxjs

### PR DESCRIPTION
**Pull request by MateuszWasik**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,49 +1,76 @@
...
 import About from './components/AboutComponent/AboutComponent'
 import Contact from './components/ContactComponent/ContactComponent'
 import UndefinedRoute from './components/UndefinedRouteComponent/UndefinedRouteComponent'
-import {BrowserRouter as Router, Route, Switch} from 'react-router-dom'
+import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'
+import { getCurrencyValues } from './services/CurrencyService'
+import { from } from 'rxjs'
 
-const { Header, Content } = Layout;
+const {Header, Content} = Layout
```
Comment (skrzepij) [inline src/App.tsx:14]: spacing

File: src/App.tsx [hunk 2]
```diff
@@ -1,49 +1,76 @@
...
+
+const resource = getCurrencyValues()
+const $observable = from(resource)
 
 const App: React.FC = () => {
+
+  const [apiRates, setApiRates] = useState<ApiRate[]>([])
+  const [date, setCurrentDate] = useState('')
+  const [dataReady, setDataReady] = useState(false)
```
Comment (skrzepij) [inline src/App.tsx:28]: can we parameterized this hooks please?

File: src/App.tsx [hunk 3]
```diff
@@ -1,49 +1,76 @@
...
+
+  useEffect(() => {
+    $observable.subscribe((response) => {
```
Comment (skrzepij) [inline src/App.tsx]: type of response? ;>

**Review by skrzepij — APPROVED** [review]

File: src/App.tsx [hunk 4]
```diff
@@ -1,49 +1,76 @@
...
-import {BrowserRouter as Router, Route, Switch} from 'react-router-dom'
+import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'
+import { getCurrencyValues } from './services/CurrencyService'
+import { from } from 'rxjs'
 
-const { Header, Content } = Layout;
+const {Header, Content} = Layout
+
+const resource = getCurrencyValues()
+const $observable = from(resource)
```
[the remark this answers is not in the document]
Comment (MateuszWasik) [reply src/App.tsx]: changes

File: src/App.tsx (hunk 2 above)
Comment (MateuszWasik) [reply src/App.tsx:28]: Done, thanks

File: src/App.tsx (hunk 1 above)
Comment (MateuszWasik) [reply src/App.tsx:14]: Fixed

File: src/App.tsx (hunk 3 above)
Comment (MateuszWasik) [reply src/App.tsx]: Added

**Pull request merged**